### PR TITLE
Resolve remaining type errors on desktop

### PR DESCRIPTION
**Pull request by captainsafia**
Alright! I'm getting no more type errors when I run `npx tsc -b applications/desktop/` on local off this branch.

### CODE REVIEW

**Comment by rgbkrk** [comment]
🎉

File: applications/desktop/src/notebook/epics/close-notebook.ts [hunk 1]
```diff
@@ -82,7 +91,15 @@ export const closeNotebookEpic = (
...
               action$.pipe(
-                ofType(
+                ofType<
+                  | CloseNotebook
+                  | coreActions.KillKernelFailed
+                  | coreActions.KillKernelSuccessful,
+                  ReturnType<
+                    | typeof coreActions.killKernelSuccessful
+                    | typeof coreActions.killKernelFailed
+                  >
```
Comment (captainsafia) [inline applications/desktop/src/notebook/epics/close-notebook.ts:101]: This looks weird and is mostly a workaround to get the redux-observable to correctly narrow down the types in downstream clauses when filtering with `ofType`. [This issue](https://github.com/redux-observable/redux-observable/issues/437) has more info.

File: packages/actions/src/actions/kernels.ts [hunk 2]
```diff
@@ -274,7 +274,7 @@ export function deleteConnectionFileSuccessful(payload: {
 }
 
 export function shutdownReplySucceeded(payload: {
-  text: string;
+  content: { restart: boolean };
   kernelRef: KernelRef;
```
Comment (rgbkrk) [inline packages/actions/src/actions/kernels.ts:278]: Are there reducers that are going to expect and use these?

Comment (captainsafia) [reply packages/actions/src/actions/kernels.ts:278]: From what I could tell, this action is only used as an intermediary message in the shutdown epic. The action that reducers process is the `KillKernelSuccessful` action which is sent when the kernel has sent a successful shutdown reply and the process has been killed.

**Pull request merged**